src/main/java/org/opentripplanner/routing/edgetype/PartialStreetEdge.java

```diff
@@ -31,8 +32,13 @@ public PartialStreetEdge(StreetEdge parentEdge, StreetVertex v1, StreetVertex v2
         super(v1, v2, geometry, name, length, parentEdge.getPermission(), false);
         setCarSpeed(parentEdge.getCarSpeed());
         this.parentEdge = parentEdge;
+
+        // Initialize length_mm from Geometry
+        if(getLength_mm() == 0) {
```

Comment:
I think it could be a good idea to explain here in a comment why the length is sometimes 0 and why it needs to be calculated here.

Reply:
Agree, also this is the only place `length_mm` is used outside its `StreetEdge`, I will move the logic to ´StreetEdge´ to iImprove encapsulation.

Reply:
I refactored this so we that there is one constructor in the TemporaryPartialStreetEdge with length and one without, and JavaDoc that explain the difference.